### DISCUSSION TITLE
How to properly simulate a controller with tinyusb

### DISCUSSION THREAD

**Discussion by amsam0** [hathach/tinyusb#1425] [Q&A]
I'm currently trying to roughly replicate the functionality of [this LUFA project](https://github.com/MrGuffels/SwitchSuperMarioTAS), instead using tinyusb and I'm not sure how to get the controller to connect and properly send packets. I've tried building on top of multiple different examples and other tinyusb projects but nothing has worked so far. I believe I've gotten the descriptors right, but the simulated controller doesn't connect or do anything. I can temporarily put the code in a repository and send the link here if needed. Thanks!

**Comment by hathach** [comment]
I guess mcu is avr since the project is baed on lufa. Unfortunately tinyusb does not support avr

**Reply by amsam0** [reply]
I have a board that supports tinyusb, I've been able to compile, flash and test the descriptors properly.

As far as I know, lufa does not support the board I have and I'm wondering if it's possible to get similar behavior to the project I linked with tinyusb. Sorry, I should've made this clear earlier.

**Reply by amsam0** [reply]
@hathach bumping this

**Reply by hathach** [reply]
To be honest, I don’t quite understand your meaning by “simulate controller”, do you mean to implement an gamepad controller ? If that is the case, there is stock example here https://github.com/hathach/tinyusb/tree/master/examples/device/hid_composite/src

Also can you be specific on which board and mcu you are using ?

**Reply by amsam0** [reply]
I’m using a raspberry pi pico (RP2040), I want to make the controller simulate a set of inputs over time, which is pre-coded.

**Reply by hathach** [reply]
OK I see, you mean hid gamepad controller, you could use gamepad example as mentioned above and sending correct button mask or reverse engineer an actual controller and follow its report payload.
